libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -108,6 +128,30 @@ private static PolicyManager createPolicyManager() {
         return new PolicyManager(serverPolicy, agentEntitlements, pluginPolicies, resolver, ENTITLEMENTS_MODULE);
     }
 
+    static void addInstrumentationMethod(
+        Map<MethodKey, CheckMethod> checkMethods,
+        Class<?> clazz,
+        String methodName,
+        Class<?> implementationClass,
+        Class<?> checkerClass,
+        String checkMethodName,
+        Class<?>... parameterTypes
+    ) throws NoSuchMethodException {
+        var instrumentationMethod = clazz.getMethod(methodName, parameterTypes);
+
+        var checkerAdditionalArguments = Modifier.isStatic(instrumentationMethod.getModifiers())
+            ? Stream.of(Class.class)
+            : Stream.of(Class.class, clazz);
+
+        var checkerArgumentTypes = Stream.concat(checkerAdditionalArguments, Arrays.stream(parameterTypes)).toArray(n -> new Class<?>[n]);
+        var additionalMethod = INSTRUMENTATION_SERVICE.lookupImplementationMethod(
+            implementationClass,
+            instrumentationMethod,
+            checkerClass.getMethod(checkMethodName, checkerArgumentTypes)
+        );
+        checkMethods.put(additionalMethod.methodToInstrument(), additionalMethod.checkMethod());
```

Comment:
This might be easier to understand if the method simply returned `additionalMethod` and the caller added it to the map. That would make this a pure function, so easier to test, reuse elsewhere, etc.

Reply:
But I agree that returning is better. And when we'll have many of those, we can do 
```
Stream.of(
   addInstrumentationMethod(...
   addInstrumentationMethod(...
).forEach(x -> checkMethods.put(x.targetMethod(), x.checkMethod());
```

Reply:
Oh interesting. I wonder if it would help if we returned a `Map.Entry`. 🤔